peachjam/models/core_document_model.py

```diff
@@ -505,10 +530,14 @@ def pre_save(self):
             self.work.save()
 
     def save(self, *args, **kwargs):
-        # give ourselves and subclasses a chance to pre-populate derived fields before saving, in case full_clean() has
-        # not yet been called
-        self.pre_save()
-        return super().save(*args, **kwargs)
+        try:
+            # give ourselves and subclasses a chance to pre-populate derived fields before saving,
+            # in case full_clean() has not yet been called
+            self.pre_save()
+            return super().save(*args, **kwargs)
+        finally:
```

Comment:
i don't think you want the try-finally block -- if pre_save() or save() fails, it won't help to call `apply_labels()`. Instead, stash the result of `super().save()` (assuming there is one) and then return it after calling `apply_labels`.